index.js

```diff
@@ -70,61 +75,44 @@ export function observeReadyElements(selector, {
 	predicate,
 } = {}) {
 	return {
-		[Symbol.asyncIterator]() {
-			const {next, complete, onCleanup, iterator} = createDeferredAsyncIterator();
-
-			function handleMutations(mutations) {
-				for (const {addedNodes} of mutations) {
-					for (const element of addedNodes) {
-						if (element.nodeType !== 1 || !element.matches(selector) || (predicate && !predicate(element))) {
-							continue;
-						}
+		async * [Symbol.asyncIterator]() {
+			const iterator = domMutations(target, {childList: true, subtree: true})[Symbol.asyncIterator]();
 
-						// When it's ready, only stop if requested or found
-						if (isDomReady(target) && element) {
-							next(element);
-							continue;
-						}
-
-						let current = element;
-						while (current) {
-							if (!waitForChildren || current.nextSibling) {
-								next(element);
-								continue;
-							}
-
-							current = current.parentElement;
-						}
-					}
-				}
+			if (stopOnDomReady) {
+				target.addEventListener('DOMContentLoaded', () => {
```

Comment:
Should also check isDomReady?

Reply:
That's not exactly relevant to this PR. If I do, these tests fail: https://github.com/sindresorhus/element-ready/blob/7917f35cabdbea87bf17c4a5c2f9dc0c6b2db96c/test.js#L248-L319

Reply:
So the code needs to be adjusted further, right?

What happens if I run this?

```js
await domLoaded;
observeReadyElements('*', {stopOnDomReady: true});
```

`elementReady` will any elements matching the selector, or stop if none are found.

`observeReadyElements` will instead continue forever, disregarding `stopOnDomReady: true`, right?

Reply:
> The new code has the bug, so I reported it.

The code in `main` has the same bug too.